12-deploy-python-microservice.ipynb

```diff
@@ -0,0 +1,730 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2136243d",
+   "metadata": {},
+   "source": [
+    "#### Creating a github repository.\n",
+    "\n",
+    "We want to create a github repository following this [link.](https://docs.github.com/en/repositories/creating-and-managing-repositories/quickstart-for-repositories)\n",
+    "\n",
+    "Once you have your repository, modify the above bash command to clone it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e4e4c39",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "git clone YOUR-GIT-REPOSITORY ~/smart-village-service-repository\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f1b272a9",
+   "metadata": {},
+   "source": [
+    "Now we are going to copy the `app.py` and `Containerfile` files to the git repository directory and enter the directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5d777e1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "cp ~/smartvillage-service/* ~/smart-village-service-repository/\n",
+    "cd ~/smart-village-service-repository/\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1c14f9e",
+   "metadata": {},
+   "source": [
+    "Next step is to commit and push our python script and Containerfile to our online repository. \n",
+    "If you want update the README.md file for documentation purposes now is also a good moment to do that."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9cfadd79",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "git add app.py\n",
+    "git commit -m \"app: Initial python microservice code\"\n",
+    "git push origin\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca889e1e",
+   "metadata": {},
+   "source": [
+    "## Deploying on Openshift\n",
+    "Openshift has a cool technology called source to image([s2i](https://github.com/openshift/source-to-image)). The code we pushed to our github repository can be easily imported into Openshift and the system will create, a container build similar to the build we did locally. Push it to a registry and start a deployment using that image. It will also setup all the routes and services needed to talk to the service.\n",
+    "\n",
+    "However, remember we need to pass 3 environment variables to the container after the deployment we need to update it to include them.\n",
+    "\n",
+    "1. From your [developer sandbox](https://console.redhat.com/openshift/sandbox) open the openshift view.\n",
+    "2. In the Developer view, select Add+ and click on the Import from Git tile.\n",
+    "3. On the Import from Git page, paste a link to your github project that includes the app.py.\n",
+    "4. Set the Resource type as Deployment.\n",
+    "5. Press create."
```

Comment:
After these steps, we'll need to describe how to add the `CLIENT_ID` and `CLIENT_SECRET` from the `smartvillage` secret, and the MAP_SERVER_URL from the route.

Reply:
I think we'll have them deploy the health checks from the topology view, which will suggest to them that they can set up health checks. 
![image](https://github.com/smartabyar-smartvillage/smartabyar-smartvillage-sandbox-course/assets/15114160/39129eb5-6bee-4dbf-9b11-654fed81dd80)